pkg/apphealth/status.go

```diff
@@ -0,0 +1,31 @@
+package apphealth
+
+import "time"
+
+const (
+	AppStatusUnhealthy uint8 = 0
+	AppStatusHealthy   uint8 = 1
```

Comment:
Can this just be a bool?

Reply:
this was existing code that I only moved to a more logical place, so I left as is to minimize changes, but I suppose I could update

Reply:
went ahead and made bool instead bc yeah no value in uint8 type and it wasn't a big diff

Reply:
Bonus points imo for just `type Status uint8` in the right pkg or unexported for the plumbing/porcelain approach if I’m understanding the point of the repo

Reply:
Hello! :slightly_smiling_face: 

It is the case that the app channel health is a boolean Healthy or Unhealthy- either the channel is open for communication or it is not. The reason why it would be `Unhealthy`, i.e. with the reasons you have above, would go into `Reason`. Those could be typed/well known, or we create a new `Message string ` which has the "error" message and make `Reason` well typed- following kube convention.

Reply:
Howdy, hope alls well across the pond! 🤠

could def see that being a handy way to enable deeper integration, good thinkin w the k8s interface there. hope the ref/granularity is handy as y'all build more things and see that uint8 stuff 🤘 😎 there's cases for string/interfaces, but def out of context on this thread at this point 😄 josh, lmk if you open an issue or anything on the k8s status wrapper Id be curious to see it & might chime in. sounds interesting. the iota things something I contributed on a golang proposal thread for, so its something I try to keep... idk familiarity with I suppose.

cheers 🍻

Reply:
The nature of enums is where it’s interesting to incorporate the message (have done a number of error libs w this), since enums sort of implicitly lack a msg. Interesting to get into if it were not only a comment on this thread lol